packages/gitbook/src/components/RootLayout/CustomizationRootLayout.tsx

```diff
@@ -66,14 +67,37 @@ export async function CustomizationRootLayout(props: {
                 sidebarStyles.list && `sidebar-list-${sidebarStyles.list}`,
                 'links' in customization.styling && `links-${customization.styling.links}`,
                 fontNotoColorEmoji.variable,
-                fonts[customization.styling.font].variable,
-                ibmPlexMono.variable
+                fontData.type === 'default' ? fontData.variable : 'font-custom'
             )}
         >
             <head>
                 {customization.privacyPolicy.url ? (
                     <link rel="privacy-policy" href={customization.privacyPolicy.url} />
                 ) : null}
+
+                {/* Custom font resources */}
+                {fontData.type === 'custom' && (
+                    <>
+                        {/* Font host preconnect */}
+                        <link rel="preconnect" href={GITBOOK_FONTS_URL} />
```

Comment:
I think it's better to use `ReactDOM.preconnect` (https://react.dev/reference/react-dom/preconnect), it can be called conditionnaly.

You should also use `ReactDOM.preload` (https://react.dev/reference/react-dom/preload)

Reply:
It'll make sure it puts as high as possible in the HTML parsing